pkg/framework/test/bin_path_finder.go

```diff
@@ -25,7 +26,11 @@ type BinPathFinder func(symbolicName string) (binPath string)
 // variable, derived from the symbolic name, and falls back to a default assets location when
 // this variable is not set
 func DefaultBinPathFinder(symbolicName string) (binPath string) {
-	envVar := "TEST_ASSET_" + strings.ToUpper(symbolicName)
+	punctuationPattern := regexp.MustCompile("[^A-Z0-9]+")
```

Comment:
I'd maybe rather have the code crash if the name is not valid? Rather than try to guess and do something that might be unexpected?

Reply:
This is because we're generating the environment variable from the executable name. So, if your asset-executable is called `my-asset` then you want your environment variable to be `TEST_ASSET_MY_ASSET`. The transformation is because typically you get different conventions for executable names (lower case, with dashes or underscores) than for environment variables (upper case, underscores only). In our case, we're working with an apiserver executable called `kube-apiserver`, and we want the configuration variable to be `TEST_ASSET_KUBE_APISERVER`. Bash would interpret `$TEST_ASSET_KUBE-APISERVER` as the environment variable `TEST_ASSET_KUBE` followed by the literal string `-APISERVER`.

The transformation strategy here owes something to the approach used by the spring framework, which allows you to set your configuration options using environment variables, build parameters, command line flags, and who knows what else -- all using the appropriate conventions for the medium. If this is too much magic for this context, we could certainly remove everything except the translation from `-` to `_`. Or we could insist that all test binaries have names which correspond to valid bash environment variables.